source/iterator/cdc_iterator.go

```diff
@@ -232,15 +236,14 @@ func (w *CDCIterator) createRecord(entry CacheEntry, reader *storage.Reader) (sd
 		return sdk.Record{}, err
 	}
 
-	return sdk.Record{
-		Metadata: map[string]string{
-			"content-type": reader.Attrs.ContentType,
+	return sdk.Util.Source.NewRecordCreate(
+		p,
+		map[string]string{
+			MetadataContentType: reader.Attrs.ContentType,
 		},
-		Position:  p,
-		Payload:   sdk.RawData(rawBody),
-		Key:       sdk.RawData(entry.key),
-		CreatedAt: reader.Attrs.LastModified,
-	}, nil
+		sdk.RawData(entry.key),
+		sdk.RawData(rawBody),
+	), nil
```

Comment:
Just to confirm: we have creates and deletes in GCS, but no updates?

Reply:
we do detect updates, but they will be considered as creates, because GCS deletes the old record and creates a new one for updates